packages/adblocker-webextension/src/index.ts

```diff
@@ -128,19 +132,74 @@ export function updateResponseHeadersWithCSP(
   return { responseHeaders };
 }
 
+const READABLE_MIME_TYPES = new Set([
+  'application/javascript',
+  'application/json',
+  'application/mpegurl',
+  'application/vnd.api+json',
+  'application/vnd.apple.mpegurl',
+  'application/vnd.apple.mpegurl.audio',
+  'application/x-javascript',
+  'application/x-mpegurl',
+  'application/xhtml+xml',
+  'application/xml',
+  'audio/mpegurl',
+  'audio/x-mpegurl',
+]);
+export const MAXIMUM_RESPONSE_BUFFER_SIZE = 10 * 1024 * 1024;
+
+export function shouldApplyReplaceSelectors(
+  request: Request,
+  details: OnHeadersReceivedDetailsType,
+): boolean {
+  const contentTypeHeader = details.responseHeaders?.find(
+    (header) => header.name.toLowerCase() === 'content-type',
+  )?.value;
+  // Content-Length includes the length of octets which is consisted of 8 bytes with endianness
+  const contentLengthHeader = details.responseHeaders?.find(
+    (header) => header.name.toLowerCase() === 'content-length',
+  )?.value;
+  if (
+    contentLengthHeader !== undefined &&
+    parseInt(contentLengthHeader, 10) >= MAXIMUM_RESPONSE_BUFFER_SIZE
+  ) {
+    return false;
+  }
```

Comment:
```suggestion
  if (
    parseInt(
      details.responseHeaders?.find(
        (header) => header.name.toLowerCase() === "content-length"
      )?.value ?? "0",
      10
    ) >= MAXIMUM_RESPONSE_BUFFER_SIZE
  ) {
    return false;
  }
```

bit more condensed

Reply:
Updated in [`bead6e9` (#14)](https://github.com/seia-soto/adblocker/pull/14/commits/bead6e965e01aab9d95de53b1f0c5f2efbdc3385)